Ext4Pkg/Ext4Dxe/Superblock.c

```diff
@@ -305,8 +306,15 @@ Ext4OpenSuperblock (
     return EFI_OUT_OF_RESOURCES;
   }
 
-  // Note that the cast below is completely safe, because EXT4_FILE is a specialisation of EFI_FILE_PROTOCOL
-  Status = Ext4OpenVolume (&Partition->Interface, (EFI_FILE_PROTOCOL **)&Partition->Root);
+  Status = Ext4OpenVolume (&Partition->Interface, &PartitionRootProtocol);
+  if (PartitionRootProtocol) {

```

Comment:
What about `EFI_ERROR`? This is not really obvious.

Reply:
I rewrote it in https://github.com/acidanthera/audk/pull/11/commits/4fa1cc7a4864bfcc1d9f5f2091f2d77990a19e56